### ISSUE TITLE
[REQUEST]

### ISSUE THREAD

**Issue by NextdoorPsycho** [riyasy/FlyPhotos#21]
The only thing preventing me from using this 100% as an end user is GIF / APNG support etc, and file associations. would love to use this

**Comment by riyasy** [comment]
Yes. I would also wish to do that. The problem is the app is so fast only because everything is manually drawn using Directx as in games. So for GIF animation, the application would need to do that manually and cannot rely on .Net builtin components which are terribly slow. I shall try to do this when there is a ample time as it is not a simple thing to do.

**Comment by NextdoorPsycho** [comment]
What about the file association prompt. You can run a script for that on install

**Comment by NextdoorPsycho** [comment]
it would also be nice to not have the image viewer open fullscreen on the first open, that should open to the size of the image, with maybe some padding at the bottom, but thats more of a convenance thing, just opening it in fullscreen is a tad annoying when you are doing things on the screen in the first place

**Comment by riyasy** [comment]
Just clicking outside of the image can exit the full screen.
Starting up full screen can be made an option in settings screen in a future version.

**Comment by riyasy** [comment]
> What about the file association prompt. You can run a script for that on install

I don't know which filetypes Fly will be able to open as it depends on the WIC (Windows imaging component) capable codecs installed on the PC. You can see all WIC capable file types in Fly's settings dialog. If the user adds a codec or uninstalls a codec, this can change. (PSD's are supported in FLY itself regardless of a WIC codec available)
For frequent filetypes which you use, you can use "Always open with" option in windows.

**Comment by NextdoorPsycho** [comment]
Would love to see the non-fullscreen startup :)

**Comment by riyasy** [comment]
GIF and APNG support added as per request in 2.4.0.
Please check and give feedback.
For other requests, please raise discrete tickets if already not raised.

**Comment by riyasy** [comment]
GIF and APNG support added as per request in 2.4.0.
Please check and give feedback.
For other requests, please raise discrete tickets if already not raised.
